### PR TITLE
feat: Implement prefix

### PR DESCRIPTION
**Pull request by theovgl**

### CODE REVIEW

**Comment by theovgl** [comment]
Ok, je dois corriger les tests en fait ca marche qu'en local ce que j'ai fait

File: src/Message.cpp [hunk 1]
```diff
@@ -107,3 +107,16 @@ const std::string	&Message::updateInputFromFields(void)
 	}
 	return this->input;
 }
+
+std::string	*Message::updatePrefixFromPeer(void)
+{
+	if (this->prefix != NULL)
+		this->prefix->clear();
+	else
+		this->prefix = new std::string;
```
Comment (valfur03) [inline src/Message.cpp:116]: Possible memory leaks if old prefix is not deleted

File: src/Message.cpp [hunk 2]
```diff
@@ -107,3 +107,16 @@ const std::string	&Message::updateInputFromFields(void)
...
+	if (this->peer.getUsername().empty())
+		*this->prefix = this->peer.getNickname() + "@" + this->peer.getStrAddr(); 
+	else
+		*this->prefix = this->peer.getNickname() + "!" + this->peer.getUsername() + "@" + this->peer.getStrAddr(); 
+	return this->prefix;
+}
```
Comment (valfur03) [inline src/Message.cpp:122]: I think the function should call the `updateInputFromFields` method

File: src/Peer.cpp [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #include "Peer.hpp"
+#include <memory>
```
Comment (valfur03) [inline src/Peer.cpp:2]: Misplaced include (should be in the header file)

File: test/Message.test.cpp [hunk 4]
```diff
@@ -97,4 +97,23 @@ TEST_CASE("Message::parse()")
...
+	SECTION("Should set the prefix with registration")
+	{
+		struct sockaddr address;
+		Peer	peer(3, address);
+		peer.registration("testUser", "", "Real Name");
+		peer.setNickname("nickTest");	
+		Message IM(peer, "command arg0");
+		IM.updatePrefixFromPeer();
+		REQUIRE(IM.prefix[0] == "nickTest!testUser@0.0.0.0");
+	}
```
Comment (valfur03) [inline test/Message.test.cpp:118]: The `struct sockaddr` fields should be set as they are really being used (e.g. the IP field)

**Review by valfur03 — CHANGES_REQUESTED** [review]

**Review by ppiques — COMMENTED** [review]
LGTM besides what @valfur03 underlined

**Pull request closed without merging**